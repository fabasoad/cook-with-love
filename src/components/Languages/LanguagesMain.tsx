import * as React from 'react'
import { useTranslation } from 'react-i18next'
import LanguageTextItem from './LanguageTextItem'

export default function LanguagesMain() {
  const { t } = useTranslation()
  return (
    <div id="languages" className="light-component">
      <div className="section-title text-center center">
        <h2>{t('cook-with-love-languages-title')}</h2>
        <hr />
      </div>
      <div className="container">
        <div className="row">
          <div className="language-list col text-center">
            <LanguageTextItem code="jp" text="JLPT N5" />
            <LanguageTextItem code="gb" text={t('cook-with-love-languages-basic')} />
            <LanguageTextItem code="ru" text={t('cook-with-love-languages-native')} />
            <LanguageTextItem code="ua" text={t('cook-with-love-languages-native')} />
          </div>
        </div>
      </div>
    </div>
  )
}
