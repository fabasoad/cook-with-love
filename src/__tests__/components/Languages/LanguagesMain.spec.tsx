/// <reference types="jest" />
import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import LanguagesMain from '../../../components/Languages/LanguagesMain'
import { useTranslation } from '../../__mocks__/react-i18next'

let tMock

beforeAll(() => {
  tMock = useTranslation().t
})

test('should render LanguageItem correctly', () => {
  const wrapper: ShallowWrapper = shallow(<LanguagesMain />)
  expect(wrapper).toMatchSnapshot()
  expect(tMock).toHaveBeenCalledTimes(4)
  expect(tMock).toHaveBeenNthCalledWith(1, 'cook-with-love-languages-title')
  expect(tMock).toHaveBeenNthCalledWith(2, 'cook-with-love-languages-basic')
  new Array<number>(3, 4).forEach(
    (n) => expect(tMock).toHaveBeenNthCalledWith(
      n, 'cook-with-love-languages-native'
    )
  )
})
