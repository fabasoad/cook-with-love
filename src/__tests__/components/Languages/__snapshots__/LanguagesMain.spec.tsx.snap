// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LanguageItem correctly 1`] = `
<div
  className="light-component"
  id="languages"
>
  <div
    className="section-title text-center center"
  >
    <h2>
      cook-with-love-languages-title
    </h2>
    <hr />
  </div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="language-list col text-center"
      >
        <LanguageTextItem
          code="jp"
          text="JLPT N5"
        />
        <LanguageTextItem
          code="gb"
          text="cook-with-love-languages-basic"
        />
        <LanguageTextItem
          code="ru"
          text="cook-with-love-languages-native"
        />
        <LanguageTextItem
          code="ua"
          text="cook-with-love-languages-native"
        />
      </div>
    </div>
  </div>
</div>
`;
